### PR TITLE
Removes support for Hadoop 2

### PR DESCRIPTION
**Pull request by tejaswini-imply**
**Release note**
Hadoop 2 support is formally deprecated in Druid 27.0.0. Hadoop 2 support is now terminated.

### CODE REVIEW

File: examples/quickstart/tutorial/hadoop3/docker/hdfs-site.xml
Comment (abhishekagarwal87) [inline examples/quickstart/tutorial/hadoop3/docker/hdfs-site.xml:1]: did we move these files or just simply deleted them?

File: docs/configuration/index.md [hunk 1]
```diff
@@ -1534,7 +1534,7 @@ Additional peon configs include:
...
-|`druid.indexer.task.defaultHadoopCoordinates`|Hadoop version to use with HadoopIndexTasks that do not request a particular version.|org.apache.hadoop:hadoop-client:2.8.5|
+|`druid.indexer.task.defaultHadoopCoordinates`|Hadoop version to use with HadoopIndexTasks that do not request a particular version.|org.apache.hadoop:hadoop-client-api:3.3.6,org.apache.hadoop:hadoop-client-runtime:3.3.6|
```
Comment (ektravel) [inline docs/configuration/index.md:1537]:
```suggestion
|`druid.indexer.task.defaultHadoopCoordinates`|Hadoop version to use with HadoopIndexTasks that do not request a particular version.|org.apache.hadoop:hadoop-client-api:3.3.6, org.apache.hadoop:hadoop-client-runtime:3.3.6|
```

File: docs/configuration/index.md [hunk 2]
```diff
@@ -1605,7 +1605,7 @@ then the value from the configuration below is used:
 |`druid.worker.numConcurrentMerges`|Maximum number of segment persist or merge operations that can run concurrently across all tasks.|`druid.worker.capacity` / 2, rounded down|
 |`druid.indexer.task.baseDir`|Base temporary working directory.|`System.getProperty("java.io.tmpdir")`|
 |`druid.indexer.task.baseTaskDir`|Base temporary working directory for tasks.|`${druid.indexer.task.baseDir}/persistent/tasks`|
-|`druid.indexer.task.defaultHadoopCoordinates`|Hadoop version to use with HadoopIndexTasks that do not request a particular version.|org.apache.hadoop:hadoop-client:2.8.5|
+|`druid.indexer.task.defaultHadoopCoordinates`|Hadoop version to use with HadoopIndexTasks that do not request a particular version.|org.apache.hadoop:hadoop-client-api:3.3.6,org.apache.hadoop:hadoop-client-runtime:3.3.6|
```
Comment (ektravel) [inline docs/configuration/index.md:1608]:
```suggestion
|`druid.indexer.task.defaultHadoopCoordinates`|Hadoop version to use with HadoopIndexTasks that do not request a particular version.|org.apache.hadoop:hadoop-client-api:3.3.6,org.apache.hadoop:hadoop-client-runtime:3.3.6|
```
```suggestion
|`druid.indexer.task.defaultHadoopCoordinates`|Hadoop version to use with HadoopIndexTasks that do not request a particular version.|org.apache.hadoop:hadoop-client-api:3.3.6, org.apache.hadoop:hadoop-client-runtime:3.3.6|
```

File: docs/operations/other-hadoop.md [hunk 3]
```diff
@@ -89,7 +89,7 @@ classloader.
 2. Batch ingestion uses jars from `hadoop-dependencies/` to submit Map/Reduce jobs (location customizable via the
 `druid.extensions.hadoopDependenciesDir` runtime property; see [Configuration](../configuration/index.md#extensions)).
 
-`hadoop-client:2.8.5` is the default version of the Hadoop client bundled with Druid for both purposes. This works with
+`hadoop-client-api:3.3.6, hadoop-client-runtime:3.3.6` is the default version of the Hadoop client bundled with Druid for both purposes. This works with
```
Comment (ektravel) [inline docs/operations/other-hadoop.md:92]:
```suggestion
The default version of the Hadoop client bundled with Druid is 3.3.6. This works with
```

**Review by ektravel — COMMENTED** [review]
Left a few suggestions for the docs.

File: examples/quickstart/tutorial/hadoop3/docker/hdfs-site.xml
Comment (tejaswini-imply) [reply examples/quickstart/tutorial/hadoop3/docker/hdfs-site.xml:1]: deleted original `tutorial/hadoop/docker/*` files and moved `tutorial/hadoop3/docker/*` files to `tutorial/hadoop/docker/` folder

**Comment by tejaswini-imply** [comment]
Thanks @abhishekagarwal87, @ektravel for the review. I have addressed your comments in the latest commit.

**Review by abhishekagarwal87 — APPROVED** [review]

**Review by ektravel — APPROVED** [review]
Docs look good.

**Review by clintropolis — APPROVED** [review]

**Pull request merged**